pykokkos/core/parsers/parser.py

```diff
@@ -123,19 +124,165 @@ def get_entities(self, style: PyKokkosStyles) -> Dict[str, PyKokkosEntity]:
+
+                        if update_type in primitives_supported:
+                            arg_obj.annotation = ast.Name(id=update_type, ctx=ast.Load())
+                                                        
+                        elif "View" in update_type:
+                            # update_type = View1D:double
+                            view_type, dtype = update_type.split(':')
+
+                            arg_obj.annotation = ast.Subscript(
+                                value = ast.Attribute(
+                                    value = ast.Name(id="pk", ctx=ast.Load()),
+                                    attr = view_type,
+                                    ctx = ast.Load()
+                                ),
+                                slice = ast.Attribute(
+                                    value = ast.Name(id="pk", ctx=ast.Load()),
+                                    attr = dtype,
+                                    ctx = ast.Load()
+                                ),
+                                ctx = ast.Load()
+                            )
+
+                        elif "Acc" in update_type:
+
+                            dtype = update_type.split(":")[1]
+
+                            arg_obj.annotation = ast.Subscript(
+                                    value = ast.Attribute(
+                                        value = ast.Name(id="pk", ctx=ast.Load()),
+                                        attr = "Acc",
+                                        ctx = ast.Load()
+                                ),
+                                slice = ast.Name(id = dtype, ctx = ast.Load()),
+                                ctx = ast.Load()
+                            )
+                        
+                        elif "pk.TeamMember" in update_type:
+
+                            arg_obj.annotation = ast.Attribute(
+                                value = ast.Name(id = "pk", ctx = ast.Load()),
+                                attr = "TeamMember",
+                                ctx = ast.Load()
+                            )
+                        else:
+                            raise ValueError("ERROR: Unsupported type inference")
+                        
+                break
+
+        return entity_tree
+
+
+    def fix_viewlayout(self, node : ast.AST, layout_change: Dict[str, str]):
+        '''
+        node: ast object for the workunit
+        layout_change: Dict that stores view variable identifier as keys against the layout type
+
+        This function returns the modified workunit ast with corrent layout decorators
+        '''
+        if len(node.decorator_list) and isinstance(node.decorator_list[0], ast.Call):
+            call_obj = node.decorator_list[0]
+            for keyword_obj in call_obj.keywords:
+                if keyword_obj.arg in layout_change:
+                    # user provided do not modify
+                    del layout_change[keyword_obj.arg]
```

Comment:
This block of code is hard to understand. Can you add some comments?

Reply:
Moved this to a method `filter_layout_change` so it can be better documented and read